DiscordAlertsTrader/brokerages/ibkr_api.py

```diff
@@ -0,0 +1,248 @@
+            order_info = {
+                'status': status,
+                'quantity': order.totalQuantity,
+                'filledQuantity': order.filledQuantity,
+                'price':trade.orderStatus.avgFillPrice,
+                'orderStrategyType': 'SINGLE',
+                "order_id" : order.orderId,
+                "orderId": order.orderId,
+                "stopPrice": order.auxPrice if order.orderType == 'STP' else None,
+                'orderType':  order.orderType,
+                'placedTime': trade.log[0].time if trade.log else int(time.time()*1000),
+                'enteredTime': trade.fills[0].time if trade.fills else int(time.time()*1000),
+                "closeTime": trade.fills[-1].time if trade.fills else int(time.time()*1000),
+                'orderLegCollection':[{
+                    'instrument':{'symbol':trade.contract.symbol},
+                    'instruction': order.action,
+                    'quantity': order.filledQuantity,
+                }]             
+
+            }
+            formatted_order = self.format_order(order_info)
+            orders_inf.append(formatted_order)
+
+        acc_inf['securitiesAccount']['orderStrategies'] = orders_inf
+        # print(acc_inf)
+        return acc_inf
+
+    def format_order(self, order:dict):
+        """ output format for order_response.Order, mimicks the order_info from TDA API"""
+        stopPrice= order['stopPrice']
+        timestamp = order['placedTime']/1000 if order['placedTime'] else None
+        enteredTime = datetime.fromtimestamp(timestamp).strftime("%Y-%m-%dT%H:%M:%S+00") if timestamp else None
+        if 'closeTime' in order:
+            timestamp = order['closeTime']/1000 if order['closeTime'] else None
+            closeTime = datetime.fromtimestamp(timestamp).strftime("%Y-%m-%dT%H:%M:%S+00") if timestamp else None
+        else:
+            closeTime = enteredTime
+        status = order['status'].upper().replace('FILLED','FILLED').replace('SUBMITTED','WORKING').replace('CANCELLED','CANCELED')
+        order_info = {
+            'status': status,
+            'quantity': order['quantity'],
+            'filledQuantity': order['filledQuantity'],
+            'price':order['price'],
+            'orderStrategyType': 'SINGLE',
+            "order_id" : order['orderId'],
+            "orderId": order['orderId'],
+            "stopPrice": stopPrice if stopPrice else None,
+            'orderType':  order['orderType'],
+            'enteredTime': enteredTime,
+            "closeTime": closeTime,
+            'orderLegCollection':[{
+                'instrument':{'symbol':order['orderLegCollection'][0]['instrument']['symbol']},
+                'instruction': order['orderLegCollection'][0]['instruction'],
+                'quantity': order['orderLegCollection'][0]['quantity'],
+            }]             
+        }
+        return order_info
+    
+    def get_quotes(self, symbol:list):
```

Comment:
please add get quotes, for example request quotes for symbols = ["NFLX_051724C450", "AMD_051724C100"]

Reply:
maybe ibkr format convention does not match the bot formatting, like tradestation API, that would require something like that to map in and out the option formats:

    def _convert_option_fromts(self, ticker):
        """
        Convert ticker from 'SPX yearmonthdayC110' to 'SPX_monthdayyearC110' format.
        
        Parameters:
        ticker (str): Ticker in the original format.
        
        Returns:
        str: Ticker in the desired format.
        """
        if " " not in ticker:
            return ticker
        symb, date_part = ticker.split()  # Split the ticker by spaces
        formatted_date = date_part[2:6] +date_part[:2]   # Reformat the date part
        return f"{symb}_{formatted_date}{date_part[6:]}"  # Combine the parts in the desired format
    
    def _convert_option_tots(self, ticker):
            """
            Convert ticker from 'SPX_monthdayyearC110' to 'SPX yearmonthdayC110' format.
            
            Parameters:
            ticker (str): Ticker in the original format.
            
            Returns:
            str: Ticker in the desired format.
            """
            if "_" not in ticker:
                return ticker
            symb, date_part = ticker.split("_")  # Split the ticker by spaces
            formatted_date = date_part[4:6] + date_part[:4]   # Reformat the date part
            return f"{symb} {formatted_date}{date_part[6:]}"  # Combine the parts in the desired format

Reply:
Done